mm2src/coins_activation/src/l2/init_l2.rs

```diff
@@ -199,10 +204,15 @@ where
         )
         .await?;
 
-        lp_register_coin(&self.ctx, coin.into(), RegisterCoinParams {
-            ticker: self.ticker.clone(),
-        })
-        .await?;
+        if let Ok(c_ctx) = CoinsContext::from_ctx(&self.ctx) {
+            let coin = coin.into();
+            c_ctx
+                .add_l2(coin.clone())
+                .await
+                .map_err(|_| RegisterCoinError::CoinIsInitializedAlready {
```

Comment:
You can replace `From<RegisterCoinError>` with `From<CoinIsAlreadyActivatedErr>` to avoid this manual conversion.